#VRML_SIM R2020a utf8
WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation 0.4828310468521281 -0.8648743763716217 -0.13735535406920513 5.647628993743788
  position 1.1131179855887754 0.7171432071275865 1.0905968502340297
  follow "MiniNezha"
  followSmoothness 0
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0.27 0.49 0
  rotation 0 0 1 0
  children [
    GPS {
    }
    Gyro {
    }
    InertialUnit {
      name "inertial_unit"
    }
    DEF LEFT_LEG Transform {
      translation 0 0 -0.125
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
          }
          device [
            PositionSensor {
              name "left_hip_encoder"
            }
            RotationalMotor {
              name "left_hip_motor"
            }
          ]
          endPoint DEF hip_link Solid {
            rotation 0 0 1 0
            children [
              DEF Hip_motor_shape Transform {
                rotation 1 0 0 1.57
                children [
                  Shape {
                    geometry Cylinder {
                      height 0.03
                      radius 0.02
                    }
                  }
                ]
              }
              DEF Hip_link Transform {
                translation 0 -0.11 0
                children [
                  Shape {
                    appearance PBRAppearance {
                      baseColor 0 0 0
                      roughness 0.5
                      metalness 0.5
                    }
                    geometry Box {
                      size 0.02 0.22 0.02
                    }
                  }
                ]
              }
              Transform {
                translation 0 -0.22 0
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 0 1
                    }
                    device [
                      PositionSensor {
                        name "left_keen_encoder"
                      }
                      RotationalMotor {
                        name "left_keen_motor"
                      }
                    ]
                    endPoint Solid {
                      rotation 0 0 1 0
                      children [
                        Transform {
                          translation 0 -0.22 0
                          rotation 1 0 0 1.57
                          children [
                            DEF Motor_Shape Shape {
                              geometry Cylinder {
                                height 0.03
                                radius 0.02
                              }
                            }
                          ]
                        }
                        DEF Keen_motor_shape Transform {
                          rotation 1 0 0 1.57
                          children [
                            Shape {
                              geometry Cylinder {
                                height 0.03
                                radius 0.02
                              }
                            }
                          ]
                        }
                        DEF Keen_link Transform {
                          translation 0 -0.11 0
                          children [
                            Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 0
                                roughness 0.5
                                metalness 0.5
                              }
                              geometry Box {
                                size 0.02 0.22 0.02
                              }
                            }
                          ]
                        }
                        Transform {
                          translation 0 -0.11 0
                          children [
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                axis 0 0 1
                                anchor 0 -0.11 0
                              }
                              device [
                                DEF left_wheel_encoder PositionSensor {
                                  name "left_wheel_encoder"
                                }
                                RotationalMotor {
                                  name "left_wheel_motor"
                                }
                              ]
                              endPoint Solid {
                                translation 0 -0.11 -0.025
                                rotation 1 0 0 1.57
                                children [
                                  DEF wheel Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0 0
                                      roughness 0.5
                                      metalness 0.5
                                    }
                                    geometry DEF wheel Cylinder {
                                      height 0.03
                                      radius 0.05
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  density -1
                                  mass 1
                                }
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE Keen_link
                      physics Physics {
                        density -1
                        mass 0.5
                        centerOfMass [
                          0 0 0
                        ]
                      }
                    }
                  }
                ]
              }
              Shape {
              }
            ]
            boundingObject USE Hip_link
            physics Physics {
              density -1
              mass 0.5
              centerOfMass [
                0 0 0
              ]
            }
          }
        }
      ]
    }
    DEF RIGHT_LEG Transform {
      translation 0 0 0.125
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
          }
          device [
            PositionSensor {
              name "right_hip_encoder"
            }
            RotationalMotor {
              name "right_hip_motor"
            }
          ]
          endPoint DEF hip_link Solid {
            translation 2.802160948195855e-06 6.012470754124876e-06 2.599882749587667e-06
            rotation -0.9239094562772463 0.3826111820102284 -1.4896141182920551e-06 1.167247058604146e-06
            children [
              USE Hip_motor_shape
              DEF Hip_link Transform {
                translation 0 -0.11 0
                children [
                  Shape {
                    appearance PBRAppearance {
                      baseColor 0 0 0
                      roughness 0.5
                      metalness 0.5
                    }
                    geometry Box {
                      size 0.02 0.22 0.02
                    }
                  }
                ]
              }
              Transform {
                translation 0 -0.22 0
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 0 1
                    }
                    device [
                      PositionSensor {
                        name "right_keen_encoder"
                      }
                      RotationalMotor {
                        name "right_keen_motor"
                      }
                    ]
                    endPoint Solid {
                      rotation 0 0 1 0
                      children [
                        Transform {
                          translation 0 -0.22 0
                          rotation 1 0 0 1.57
                          children [
                            USE Motor_Shape
                          ]
                        }
                        USE Keen_motor_shape
                        DEF Keen_link Transform {
                          translation 0 -0.11 0
                          children [
                            Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 0
                                roughness 0.5
                                metalness 0.5
                              }
                              geometry Box {
                                size 0.02 0.22 0.02
                              }
                            }
                          ]
                        }
                        Transform {
                          translation 0 -0.11 0
                          children [
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                axis 0 0 1
                                anchor 0 -0.11 0
                              }
                              device [
                                PositionSensor {
                                  name "right_wheel_encoder"
                                }
                                RotationalMotor {
                                  name "right_wheel_motor"
                                }
                              ]
                              endPoint Solid {
                                translation 0 -0.11 0.025
                                rotation 1 0 0 1.5699999999999996
                                children [
                                  DEF wheel Shape {
                                    appearance PBRAppearance {
                                      baseColor 1 0 0
                                      roughness 0.5
                                      metalness 0.5
                                    }
                                    geometry DEF wheel Cylinder {
                                      height 0.03
                                      radius 0.05
                                    }
                                  }
                                ]
                                boundingObject USE wheel
                                physics Physics {
                                  density -1
                                  mass 1
                                }
                              }
                            }
                          ]
                        }
                      ]
                      boundingObject USE Keen_link
                      physics Physics {
                        density -1
                        mass 0.5
                        centerOfMass [
                          0 0 0
                        ]
                      }
                    }
                  }
                ]
              }
              Shape {
              }
            ]
            name "solid(1)"
            boundingObject USE Hip_link
            physics Physics {
              density -1
              mass 0.5
              centerOfMass [
                0 0 0
              ]
            }
          }
        }
      ]
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        roughness 0.1
        metalness 0.8
      }
      geometry Box {
        size 0.15 0.1 0.23
      }
    }
  ]
  name "MiniNezha"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 4
    centerOfMass [
      0 0 0
    ]
  }
  controller "my_controller_python"
}
